.github/workflows/test.yml

```diff
@@ -0,0 +1,39 @@
+name: test
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  test:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        go-version: [1.12.x, 1.19.x]
```

Comment:
Unit tests are running on both 1.12 (oldest supported version) and 1.19 (current release)

Reply:
Sounds good. If we run into issues supporting older versions of `go`, then I'd be very okay with dropping support for older versions... but until then fine if we keep them around, and definitely nice to have CI checking.

Reply:
Yes, I had the same train of thought. Initially thought "let's drop it", but it (currently) didn't seem like a massive burden to maintain, so decided to keep it (for now).